### ISSUE TITLE
PRM model training details

### ISSUE THREAD

**Issue by bdiy90** [openai/prm800k#11]
hello, I have a doubt with the training of the PRM using the dataset from the paper.

I've seen two related issues: #10 and #7 

from #10 I understood you create one sequence for each intermediate trajectory and you compute a single reward at the end of each sequence correspodning to the last step in the sequence

using the same example 
```
Step 1: [A, B, C] Chosen: A
Step 2: [D] Chosen: D
Step 2: [E, F] Chosen: E

Training sequences with labels:
[A,1] [B,-1] [C,-1] [A, D, 1] [A, D, E, 1] [A, D, F, -1]
```

but in issue #7 I understand the correct way of doing is not to have a sequence for each intermediate trajectory, but instead predict reward after each step in a larger trajectory:

```
Training sequences:
[B, -1]
[C, -1]
[A, 1, D, 1, E 1]
[A, 1, D, 1, F, -1]
```

this way the number of training sequeunces is reduced.

which way did you use?

**Comment by Huntrr** [comment]
note [B] could still be +1 even though it wasn't chosen (sometimes multiple steps were +1 but we could only choose one to continue with). also we rated sequences -1, 0, +1, so any of these scores could also have been 0

we did the first way.

the second way is more complicated. you probably want to make sure when the model is predicting the score for `E` that it can't see the `A=1, D=1` since that gives it a big hint when it's trying to guess the score for `E`
